service/worker/batcher/workflow.go

```diff
@@ -132,8 +133,11 @@ func BatchWorkflow(ctx workflow.Context, batchParams BatchParams) error {
 }
 
 func setDefaultParams(params BatchParams) (BatchParams, error) {
-	if params.BatchType == "" || params.Reason == "" || params.DomainName == "" || params.QueryCondition == "" {
-		return BatchParams{}, fmt.Errorf("must provide required parameters: BatchType/Reason/DomainName/QueryCondition")
+	if params.BatchType == "" ||
```

Comment:
It will be nice to split setDefault and validation.. Call `setDefault` first.

And then call validate which returns an error.

Reply:
OK